src/main/java/edu/utah/ece/async/sboldesigner/sbol/editor/Part.java

```diff
@@ -122,16 +122,44 @@ public Image getImage() {
 		return smallImage;
 	}
 
+
 	/**
 	 * Creates a CD in design using roles.
 	 */
-	public ComponentDefinition createComponentDefinition(SBOLDocument design) {
+	public ComponentDefinition createComponentDefinition(SBOLDocument design) 
+	{
+		return createComponentDefinition(design, ComponentDefinition.DNA);
+	}
+	
+	/**
+	 * Create a ComponentDefinition that will set the roleType base on the given ComponentDefinition Type.
+	 * @param design - The SBOL document that the ComponentDefinition will be created in.
+	 * @param compDefType - The type that the ComponentDefinition will be set to.
+	 * @return The ComponentDefinition that was created.
+	 */
+	public ComponentDefinition createComponentDefinition(SBOLDocument design, URI compDefType) {
 		// change list of roles to set of roles
 		Set<URI> setRoles = new HashSet<URI>(roles);
+		
 		// create ComponentDefinition using the following parameters
-		try {
+		try 
+		{
 			String uniqueId = SBOLUtils.getUniqueDisplayId(null, getDisplayId(), "1", "CD", design);
-			ComponentDefinition comp = design.createComponentDefinition(uniqueId, "1", ComponentDefinition.DNA);
+			ComponentDefinition comp = design.createComponentDefinition(uniqueId, "1", compDefType);
+			
+			if(compDefType.equals(ComponentDefinition.DNA))

```

Comment:
I don't understand all the setRoles modifications.  If it's DNA, it doesn't necessarily mean it's also always an ENGINEERED_REGION.  Same with RNA and SMALL_MOLECULE.

Could you explain?